dts/arm/atmel/samd.dtsi

```diff
@@ -33,6 +33,11 @@
 			<0x0080A048 0x4>;
 	};
 
+	aliases {
+		pinmux-a = &pinmux_a;
+		pinmux-b = &pinmux_b;
```

Comment:
Why did you add aliases for pinmux and sercom, but not gpio? The generated dts instance indices aren't guaranteed to match the static hardware indices. Consider the case that gpio port a is disabled in dts, then gpio port b will be the first enabled instance in dts and get index 0.

Reply:
I only added them because they where the only ones used.  That is the current boards only reference the pinmux rather than the GPIO devices.  The GPIO driver itself is independent of the indexing.  I have no problem adding them for GPIO however.

That is, since the GPIO drivers are self contained, the indexing doesn't matter (i.e. it doesn't matter if index 0 refers to PORT A or PORT B, since it has the correct address from the DTS).  Basically, nothing in the GPIO instance has to reference anything looked up in absolute indices (e.g. clocking) so it's not required.